var/spack/repos/builtin/packages/zlib/package.py

```diff
@@ -63,6 +67,17 @@ class Zlib(MakefilePackage, Package):
 
     license("Zlib")
 
+    @classmethod
+    def determine_version(cls, lib):
+        match = re.search(r"lib\S\.so\.(\d+)\.(\d+)\.(\d+)", lib)
+        if match:
+            ver = "{0}.{1}.{2}".format(
```

Comment:
FYI. We are encouraging use of Python f-strings (https://docs.python.org/3/tutorial/inputoutput.html) as we are also converting to them in the core when they make sense.